### DISCUSSION TITLE
Suppress auto-generated attributes of each span

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#6182] [Q&A]
Hi,
We are seeing an increase in the size (in bytes) of each span using OpenTelemetry i.e.: Earlier the span data size was in around 1KB and now it has increased by an additional 1.5x  as ~ 2.4KB and this is contributing to more disk space usage.
There is a lot of duplicate data across the spans (under the same trace Id) in Tags and Process sections on Jaeger UI.

Is there way to suppress / disable such duplicate data being attached to each span? Or solution to reduce this situation.

**Comment by trask** [comment]
this might help: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#disabling-automatic-resourceproviders

**Reply by itsmaneesh** [reply]
@trask  Thank you for the inputs. 
I tried to disable each of Resourec Provider using Resource Providers environment variable `OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS`. But, looks like, only Operating System `io.opentelemetry.sdk.extension.resources.OsResource` specific attributes are being suppressed and Process `io.opentelemetry.sdk.extension.resources.ProcessResource` and Process Runtime (Java Runtime) `io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource` attributes are STILL showing up on Jaeger UI. 
Shall I create an issue to investigate on this?

**Reply by trask** [reply]
can you share the exact value that you set for the env var?

**Reply by itsmaneesh** [reply]
For e.g.: for ProcessRuntimeResource
 OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS:  io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource 

Also tried with all three resource types as below:
OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS: io.opentelemetry.sdk.extension.resources.ProcessResource, io.opentelemetry.sdk.extension.resources.OsResourceProvider, io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource

**Reply by itsmaneesh** [reply]
@trask

**Reply by trask** [reply]
I see that there's a test for this: https://github.com/open-telemetry/opentelemetry-java/blob/852e755eebcad4c740b1de9d606812a5a470eb72/sdk-extensions/autoconfigure/build.gradle.kts#L150-L166

can you see if it works when you are using OTel Java SDK (and not the Java agent), to try to narrow down where the issue is (and which repo it belongs in)?

**Reply by itsmaneesh** [reply]
@trask We are using Java agent and I have configured the environment variable as below:
`OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS: io.opentelemetry.sdk.extension.resources.ProcessResource, io.opentelemetry.sdk.extension.resources.OsResourceProvider, io.opentelemetry.sdk.extension.resources.ProcessRuntimeResource`

However, I am still seeing the Process and ProcessRuntime resources are still being shown on the span attributes.
Do you think, the above configuration might not work with Java agent?

**Reply by mateuszrzeszutek** [reply]
Instead of `ProcessRuntimeResource`/`ProcessResource` you should mention the providers, i.e.
```
OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS: io.opentelemetry.sdk.extension.resources.ProcessResourceProvider,io.opentelemetry.sdk.extension.resources.OsResourceProvider,io.opentelemetry.sdk.extension.resources.ProcessRuntimeResourceProvider
```

**Reply by itsmaneesh** [reply]
@mateuszrzeszutek can you please point me to a document where I can find list of `ResourceProvider` that I can use them for  `OTEL_JAVA_ENABLED_RESOURCE_PROVIDERS` env variable.

**Reply by mateuszrzeszutek** [reply]
Unfortunately, there's no such document at the moment. You can find most of the resources included in the agent in this directory though: https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources

**Reply by itsmaneesh** [reply]
Thank you @mateuszrzeszutek . 
I am using Java agent with just `OTEL_JAVA_ENABLED_RESOURCE_PROVIDERS` : `io.opentelemetry.sdk.extension.resources.ProcessResourceProvider` then I was expecting to see just `process.*` attributes on Jaeger UI. But, instead, I am seeing all the attributes like `os.*`, `process.*`, `process.runtime.*`, `host.*`, `container.*`., `telemetry.*` etc.
Am I missing something here? Please advise.

**Reply by mateuszrzeszutek** [reply]
What version of the javaagent are you using?

**Reply by itsmaneesh** [reply]
@mateuszrzeszutek I am using `v1.10.0` - Java agent.

**Reply by mateuszrzeszutek** [reply]
That's a rather old version; looks like the `OTEL_JAVA_ENABLED_RESOURCE_PROVIDERS` property was introduced in v.1.13.0. Can you try using the latest javaagent and see if it works?

**Reply by itsmaneesh** [reply]
@mateuszrzeszutek  Got it. Will try this `OTEL_JAVA_ENABLED_RESOURCE_PROVIDERS` variable with `v1.13.0`.
Meanwhile, can you please let me know whether we can disable `telemetry.*` attributes? If so, how to disable `telemetry.*` attributes using Java agent v1.10.0?
I couldn't find the relevant package in this list : https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/resources/src/main/resources/META-INF/services/io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider

**Reply by mateuszrzeszutek** [reply]
No, the `telemetry.*` attributes are always added by the SDK, they're not configurable.

**Reply by itsmaneesh** [reply]
Thank you.
